libafl_libfuzzer/README.md

```diff
@@ -0,0 +1,143 @@
+Remember to `cargo update` often if using the experimental changes, and please [submit an issue]
+if you encounter problems while using `libfuzzer-best`!
+
+#### Caveats
+
+Like harnesses built with `libfuzzer-sys`, Rust targets which build other libraries (e.g. C/C++ FFI) may not
+automatically apply instrumentation.
+In addition to installing clang, you may also wish to set the following environmental variables:
+
+```bash
+CC=clang
+CXX=clang++
+CFLAGS='-fsanitize=fuzzer-no-link'
+CXXFLAGS='-fsanitize=fuzzer-no-link'
+```
+
+### Usage as a standalone library (for C/C++/etc.)
+
+The runtime for `libafl_libfuzzer` may be used standalone as a direct replacement for libFuzzer with other targets as
+well.
+To do so, [ensure a recent nightly version of Rust is installed](https://rustup.rs/), then enter the
+[`libafl_libfuzzer_runtime`](libafl_libfuzzer_runtime) folder and build the runtime with the following command:
+
+```bash
+./build.sh
+```
+
+The static library will be available at `libFuzzer.a` in the [`libafl_libfuzzer_runtime`](libafl_libfuzzer_runtime)
+directory.
+If you encounter build failures without clear error outputs that help you resolve the issue, please [submit an issue].
+
+This library may now be used in place of libFuzzer.
+To do so, change your CFLAGS/CXXFLAGS from `-fsanitize=fuzzer` to:
+
+```
+-fsanitize=fuzzer-no-link -L/path/to/libafl_libfuzzer_runtime -lFuzzer
+```
+
+Alternatively, you may directly overwrite the system libFuzzer library and use `-fsanitize=fuzzer` as normal.
+This changes per system, but on my machine is located at `/usr/lib64/clang/16/lib/linux/libclang_rt.fuzzer-x86_64.a`.
+
+#### Caveats
+
+This standalone library is _not_ compatible with Rust targets; you must instead use the crate-based dependency.
+This is due to potential symbol conflict between your harness and the fuzzer runtime, which is resolved by additional
+build steps provided in the `libafl_libfuzzer` crate itself.
+
+## Flags
+
+You can pass additional flags to the libFuzzer runtime in `cargo-fuzz` like so:
+
+```bash
+cargo fuzz run fuzz_target -- -extra_flag=1
+```
+
+When the runtime is used standalone, flags may be passed just like normal libFuzzer.
+
+You will commonly need this for flags such as `-ignore_crashes=1` and `-timeout=5`. In addition
+to partial support of libfuzzer flags, `libafl_libfuzzer` offers:
+
+- `-dedup=n`, with `n` = 1 enabling deduplication of crashes by stacktrace.
+- `-grimoire=n`, with `n` set to 0 or 1 disabling or enabling [grimoire] mutations, respectively.
+    - if not specified explicitly, `libafl_libfuzzer` will "guess" which setting is appropriate
+    - you should disable grimoire if your target is not string-like
+- `-report=n`, with `n` = 1 causing `libafl_libfuzzer` to emit a report on the corpus content.
+- `-skip_tracing=n`, with `n` = 1 causing `libafl_libfuzzer` to disable comparison log tracing.
```

Comment:
'comparison log', should be `cmplog` no? I don't think anyone knows what 'comparison log' is...